### PR TITLE
proxy: remove unused ifaces and code for proxy <-> endpoint interaction

### PR DESCRIPTION
**Pull request by mhofstetter**
This PR removes unused interfaces and code for the proxy <-> endpoint interaction.

They have been probably missed during previous proxy refactorings.

### CODE REVIEW

**Review by pippolo84 — APPROVED** [review]
Thanks! :+1:

**Review by tklauser — APPROVED** [review]

File: pkg/envoy/xds_server_test.go [hunk 1]
```diff
@@ -40,11 +40,9 @@ var (
 	Identity = identity.NumericIdentity(123)
```
Comment (tklauser) [inline pkg/envoy/xds_server_test.go]: nit: this is now unused

**Review by sayboras — APPROVED** [review]
LGTM :heavy_check_mark:

File: pkg/envoy/xds_server_test.go (hunk 1 above)
Comment (mhofstetter) [reply pkg/envoy/xds_server_test.go]: thanks. removed

**Pull request merged**